include/mtxclient/http/client.hpp

```diff
@@ -449,11 +449,14 @@ class Client : public std::enable_shared_from_this<Client>
           ErrCallback callback);
 
         //! POST a new room listing to the public rooms directory.
-        void post_public_rooms(const std::string &server, const nhlomann::json &j, 
-                                Callback<mtx::responses::PublicRooms> cb); 
+        void post_public_rooms(const mtx::requests::PublicRooms &req, 
+                                Callback<mtx::responses::PublicRooms> cb, 
+                                const std::string &server = "matrix.org"); 
         //! GET the public rooms directory listing.
-        void get_public_rooms(int limit, const std::string &since, const std::string &server,
-                                Callback<mtx::responses::PublicRooms> cb);
+        void get_public_rooms(Callback<mtx::responses::PublicRooms> cb, 
+                                const std::string &server = "matrix.org", 
+                                int limit = std::numeric_limits<int>::max(), 
```

Comment:
@deepbluev7 should limit really be of type `size_t`? Spec says it's an integer but they may not actually mean `int`

Reply:
The max value for limit is 2^53 probably, so size_t sounds right. But you probably want to make the default value 0 for no limit. And the server should default to an empty string.